src/libs/antares/study/parts/hydro/container.cpp

```diff
@@ -106,6 +106,34 @@ void PartHydro::reset()
     }
 }
 
+using setProperty = std::function<void (IniFile::Property*, Area*)>;
+
+static bool loadProperties(Study& study, IniFile::Property* property, const std::string& filename, setProperty funcToSetProperty)
+{
+    if (!property)
+        return false;
+
+    bool ret = true;
+
+    // Browse all properties
+    for (; property; property = property->next)
+    {
+        AreaName id = property->key;
+        id.toLower();
+
+        Area* area = study.areas.find(id);
+        if (area)
+        {
+            funcToSetProperty(property, area);
+        }
+        else
+        {
+            logs.warning() << filename << ": `" << id << "`: Unknown area";
+        }
+    }
+    return ret;
```

Comment:
This function always returns true. Maybe `funcToSetProperty` should return a `bool` ? Since these functions can issue an error, it would make sense.

Reply:
ret is supposed to be captured by funcToSetProperty, i agree that returning bool may be clearer